src/app/pages/table-page/batch-actions-table.component.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  CUSTOM_ELEMENTS_SCHEMA,
+  Component,
+  ElementRef,
+  ViewChild,
+  inject,
+  Renderer2,
+  AfterViewInit, Input
+} from '@angular/core'
+import {TableData} from './table-data'
+import {ModalDirective} from '../../directives/modal.directive'
+
+@Component({
+  selector: 'app-batch-actions-table',
+  template: `
+    <h1>Batch Actions</h1>
+    <tds-table enable-multiselect vertical-dividers="false" compact-design="false" enable-responsive="false">
+      <tds-table-toolbar table-title="Batch action">
+        <div slot="end">
+          <tds-button type="primary" size="sm" text="Download" (click)="download()"></tds-button>
```

Comment:
```suggestion
          <tds-button id="my-modal-button" type="primary" size="sm" text="Download"></tds-button>
```

Should work without the click event handler. The id of the button should be connected with the selector in the modal

Reply:
Could the modal fetch the content rather then it being set in the click event? or maybe the click event should only set the content, and not hide/show the modal?

Reply:
I don't want the modal to know anything on how to fetch data. A common design pattern in angular is to create "Smart/Presentational" components, where the smart knows how to fetch data and the presentational components basically only accepts data as inputs and emits events (if neccesary)

Reply:
Makes sense, I think I just got confused by the name showModal on the function. It it the same as the method exposed on our tds-modal, so I though we were calling that method instead of letting the selector work. Nice stuff

Reply:
No event when it opens at the moment, only for when it closes.

Reply:
actually i don't use the selector with button id at all. i wrapped the tds-modal in a reusable angular component and i have the "show" attribute so when i create my wrapper component and pass the content to it, it automatically opens the modal and i have data binding with the "content" property in my wrapper component.

I did it that way so i only have the <tds-modal></tds-modal> template code in one place

Reply:
Lovely dubly!